### PR TITLE
Tlportal 281 - add Canvas todolms (activity stream) feed to Dash backend

### PR DESCRIPTION
**Pull request by dlhaines**

### CODE REVIEW

File: server/canvas_api_response.rb [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+  # Store the body of the response.  All output from Canvas API goes through this class.
+  attr_accessor :body_string, :body_json
+
+  ## TODO: allow input as string or json
+
+  # Input is in string format
+  def initialize(body)
+    @body_string = body
+    @body_json = JSON.parse(body)
+    @canvas_instance_prefix = 'https://umich.test.instructure.com.dummy'
```
Comment (lsloan) [inline server/canvas_api_response.rb]: Is this the final value to be used in production?

Comment (dlhaines) [inline server/canvas_api_response.rb]: Good point. It is used for the moment, I've filed a Jira to find and use
the real value.

File: server/courselist.rb [hunk 2]
```diff
@@ -92,6 +92,9 @@ class CourseList < Sinatra::Base
   # name of application to use for CTools HTTP direct information
   config_hash[:ctools_http_application_name] = "CTQA-DIRECT"
```
Comment (zqian) [inline server/courselist.rb]: should this and the below canvas_esb_application_name be set inside the external configuration file instead?

File: server/data_provider.rb [hunk 3]
```diff
@@ -92,9 +125,10 @@ def dataProviderToDoCToolsLMS(uniqname)
 
     dataProviderInit
 
-    unless @useCtoolsHTTPToDoLMSProvider.nil?
+    unless @ctoolsHash[:ToDoLMSProvider].nil?
       logger.error "#{__method__}: #{__LINE__}: deal with status in WAPI wrapper"
-      raw_todos = @ctoolsHTTPDirectToDoLMS.(uniqname)
+      #raw_todos = @ctoolsHTTPDirectToDoLMS.(uniqname)
```
Comment (zqian) [inline server/data_provider.rb]: delete the unwanted code instead of commenting out...

File: server/courselist.rb (hunk 2 above)
Comment (dlhaines) [inline server/courselist.rb]: Yes.  I'll fix that in a latter jira.

File: server/data_provider.rb (hunk 3 above)
Comment (dlhaines) [inline server/data_provider.rb]: Good idea.

File: server/spec/test_integration_WAPI_CANVAS.rb [hunk 4]
```diff
@@ -57,10 +59,11 @@ def setup
 
     logger.level=TestHelper.getCommonLogLevel
     #logger.level=Logger::ERROR
-    #logger.level=Logger::DEBUG
+    logger.level=Logger::DEBUG
 
     #@default_application_name = 'SD-QA-CANVAS'
-    @default_application_name = 'Canvas-TL-TEST'
+    #@default_application_name = 'Canvas-TL-TEST'
```
Comment (zqian) [inline server/spec/test_integration_WAPI_CANVAS.rb]: Remove this line?

File: server/local/studentdashboard-PROD.yml [hunk 5]
```diff
@@ -1,16 +1,31 @@
...
-# PURPOSE: default settings for running Latte on a production server.
+# PURPOSE: settings for PRODUCTION instance
 #
-# Application name which must match a value in the security.yml file.  Identifies the credentials and ESB to use.
-application_name: StudentDashboard-PROD
+################## Data source configuration
+# course / term information application name (mpathways)
+application_name: SD-QA
+# ctools information application name
+ctools_http_application_name: CTQA-DIRECT
```
Comment (zqian) [inline server/local/studentdashboard-PROD.yml]: Is this still CTQA-DIRECT for prod yml file?

File: server/local/studentdashboard-PROD.yml [hunk 6]
```diff
@@ -1,16 +1,31 @@
...
+# canvas information application name
+canvas_esb_application_name: CANVAS-TL-QA
```
Comment (zqian) [inline server/local/studentdashboard-PROD.yml]: Same as above: still CANVAS-TL_QA in prod yaml file?

**Pull request merged**